internal/resources/cluster_onoff_schedule.go

```diff
@@ -0,0 +1,392 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+
+	scheduleapi "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/cluster_onoff_schedule"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &ClusterOnOffSchedule{}
+	_ resource.ResourceWithConfigure   = &ClusterOnOffSchedule{}
+	_ resource.ResourceWithImportState = &ClusterOnOffSchedule{}
+)
+
+//const errorMessageAfterOnOffScheduleCreation = "Cluster On/Off Schedule creation is successful, but encountered an error while checking the current" +
+//	" state of the cluster on/off schedule. Please run `terraform plan` after 1-2 minutes to know the" +
+//	" current on/off schedule state. Additionally, run `terraform apply --refresh-only` to update" +
+//	" the state from remote, unexpected error: "
+
+const errorMessageWhileOnOffScheduleCreation = "There is an error during cluster on/off schedule creation. Please check in Capella to see if any hanging resources" +
+	" have been created, unexpected error: "
+
+// ClusterOnOffSchedule is the OnOffSchedule resource implementation.
+type ClusterOnOffSchedule struct {
+	*providerschema.Data
+}
+
+// NewClusterOnOffSchedule is a helper function to simplify the provider implementation.
+func NewClusterOnOffSchedule() resource.Resource {
+	return &ClusterOnOffSchedule{}
+}
+
+// Metadata returns the OnOffSchedule resource type name.
+func (c *ClusterOnOffSchedule) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_onoff_schedule"
```

Comment:
should this be cluster_onoff_schedule? since the schedule can only be defined at cluster level?

CC: @rajagp